libyul/backends/evm/OptimizedEVMCodeTransform.cpp

```diff
@@ -351,7 +351,10 @@ void OptimizedEVMCodeTransform::createStackLayout(std::shared_ptr<DebugData cons
 				[&](JunkSlot const&)
 				{
 					// Note: this will always be popped, so we can push anything.
-					m_assembly.appendInstruction(evmasm::Instruction::CODESIZE);
+					if (m_assembly.evmVersion().hasPush0())
+						m_assembly.appendConstant(0);
+					else
+						m_assembly.appendInstruction(evmasm::Instruction::CODESIZE);
```

Comment:
As the comment says: we can push *anything* here - the reason why we used ``CODESIZE`` was that it's only 1 byte and 2 gas (whereas a ``PUSH1 0`` was 2 bytes and 3 gas) - if we have ``PUSH0`` (which is also 1 byte and 2 gas), that's definitely the nicer choice.